source/_posts/2016-01-25-upgrading-to-ruby-on-rails-5-0.markdown

```diff
@@ -0,0 +1,226 @@
+      minitest-rails (~> 2.1) was resolved to 2.2.0, which depends on
+        railties (~> 4.1)
+```
+
+We are using minitest-rails-capybara, which depends on minitest-rails. We just erased the Gemfile, but owing to this dependency, minitest-rails continues trying to install. We follow the same procedure from the previous case, asking ourselves, “What do we need minitest-rails-capybara for?” The reality is that we don’t need it either, we’ll simply use minitest-capybara, which is one of its dependencies.
+
+Therefore:
+
+- We substitute the minitest-capybara for the minitest-rails-capybara entry
+- We call a require of `capybara/rails` in place of `minitest/rails/capybara` in the `test_helper.rb`.
+- We make sure that the integration tests inherit from `ActionDispatch::IntegrationTest` instead of from `Capybara::Rails::TestCase` (a class of minitest-rails-capybara).
+- And lastly we add the following to `test_helper.rb` ...
+
+```
+class ActionDispatch::IntegrationTest
+  include Capybara::DSL
+  include Capybara::Assertions
+
+  ...
+end
+```
+
+We execute `bundle update rails` again and now it installs everything correctly.
+
+Now we can run the tests, for example.
+Hence, we execute `bin/rails test` (this is the new way to run tests in Rails 5!) and we have various tests failing.
+
+
+## Problem 1 - the tests fail and we don’t have any further information
+
+From this point forward, the Rails guides give us no further information and we still have various tests failing. In particular, failing is a series of tests of the following form ...
+
+```
+ActionController::RoutingError: No route matches [POST] "/accounts/1/transactions/1"
+```
+
+The next thing we can do is force our application to use [Rails master](https://github.com/rails/rails) in order to see if some of the test failures are due to problems already solved.
+So, we substitute the Gemfile Rails entry with `gem 'rails', github: 'rails/rails'` and once again execute `bundle update rails`.
+
+We return to running the tests with `bin/rails test` and the tests that fail with RoutingError disappear.
+There are, however, other tests that continue failing.
+And now, we have no choice but to begin investigating the errors one by one. 
+
+
+## Problem 2 - expected Array (got String) for param xxx
+
+One of the errors that `bin/rails test` gives us is ...
+
+```
+E
+
+ActionController::BadRequest: Invalid request parameters: expected Array (got String) for param `kind'
+
+bin/rails test test/features/entries/create_entry_test.rb:38
+```
+
+The feature test tries to create an Entry from the form. An Entry in our system is a movement of money that, among other attributes, has a kind attribute with a value of expense, income or transfer. The kind field is an enum in the model and is represented by a radio button in the form. The error indicates that for some reason Rails is expecting an Array when it should be expecting a String.
```

Comment:
I don't like the `from the form` part.

Maybe?
`The feature test tries to create an Entry using the corresponding form.`